pkg/sensors/handler.go

```diff
@@ -94,10 +94,19 @@ func (h *handler) updatePolicyFilter(tp tracingpolicy.TracingPolicy, tpID uint64
 }
 
 func (h *handler) addTracingPolicy(op *tracingPolicyAdd) error {
+
+	var namespace string
+	if tpNs, ok := op.tp.(tracingpolicy.TracingPolicyNamespaced); ok {
+		namespace = tpNs.TpNamespace()
+	} else {
+		namespace = ""
+	}
+
+	collectionKey := collection_key{op.name, namespace}
 	// allow overriding existing policy collection that resulted in an error
 	// during the loading state
-	if col, exists := h.collections[op.name]; exists && col.state != LoadErrorState {
-		return fmt.Errorf("failed to add tracing policy %s, a sensor collection with the name already exists", op.name)
+	if col, exists := h.collections[collectionKey]; exists && col.state != LoadErrorState {
+		return fmt.Errorf("failed to add tracing policy %s, a sensor collection with the name already exists", collectionKey)
```

Comment:
For pretty printing, you can define `String()` method on `collectionKey` that returns "namespace/name" for namespaced policies and just "name" for clusterwide.

Reply:
Added this in the latest commit as well. Thanks!